http/src/main/java/io/micronaut/http/filter/FilterRunner.java

```diff
@@ -76,25 +48,29 @@
  */
 @Internal
 public class FilterRunner {
-    private static final Predicate<FilterMethodContext> FILTER_CONDITION_ALWAYS_TRUE = runner -> true;
 
     /**
      * All filters to run. Request filters are executed in order from first to last, response
      * filters in the reverse order.
      */
-    private final List<GenericHttpFilter> filters;
+    private final List<InternalHttpFilter> filters;
     private final PropagatedContext initialPropagatedContext = PropagatedContext.getOrEmpty();
 
     /**
      * Create a new filter runner, to be used only once.
      *
-     * @param filters           The filters to run
+     * @param filters The filters to run
      */
     public FilterRunner(List<GenericHttpFilter> filters) {
-        this.filters = Objects.requireNonNull(filters, "filters");
+        this.filters = Objects.requireNonNull(filters, "filters").stream().map(f -> {
```

Comment:
replace with an unchecked cast for performance. This stream can never fail because GenericHttpFilter is sealed.